src/tech/v3/dataset/metamorph.clj

```diff
@@ -0,0 +1,71 @@
+(ns tech.v3.dataset.metamorph
+  (:require [tech.v3.dataset :as ds]
+            [tech.v3.dataset.modelling :as ds-mod]
+             [tech.v3.protocols.dataset :as prot]))
+
+(defn dataset?
+  "Is `ds` a `dataset` type?"
+  [ds]
+  (satisfies? prot/PColumnarDataset ds))
+
+(defmacro build-pipelined-function
+  [f m]
+  (let [args (map (comp vec rest) (:arglists m))]
+    `(defn ~(symbol (name f))
+       ~@(for [arg args
+               :let [narg (mapv #(if (map? %) 'options %) arg)
+                     [a & r] (split-with (partial not= '&) narg)]]
+           (list narg `(fn [ds#]
+                         (let [ctx# (if (dataset? ds#)
+                                      {:metamorph/data ds#} ds#)]
+                           (assoc ctx# :metamorph/data (apply ~f (ctx# :metamorph/data) ~@a ~(rest r))))))))))
+
+
+
+(def ^:private includes-dataset
+  '#{categorical->number categorical->one-hot})
+
+
+
+(defmacro ^:private process-all-api-symbols-dataset
```

Comment:
I am sure there is a way to unify this and the following macro,
but I have no clue how...
I am very new to macros.

Maybe you can do that.

Reply:
I don't think the unification is necessary.  Are you finished with this PR?

Reply:
Not sure, if we need an exchange of views regarding, which functions should be lifted.

My goal is, to have all functions needed for being able to crate machine learning pipelines.

Maybe some other from categorical.clj would be needed.

I propose to start with this PR as is, 

and if more are needed, I do other PRs